controllers/storagecluster/storageconsumer.go

```diff
@@ -284,23 +318,41 @@ func getLocalVolumeSnapshotClassNames(ctx context.Context, kubeClient client.Cli
 	}); err != nil {
 		return nil, err
 	}
+	classesSpec := internalConsumer.Spec.VolumeSnapshotClasses
+	destToSrcMap := util.GetDestToSrcClassMapping(classesSpec)
 	for idx := range volumeSnapshotClassesInCluster.Items {
-		// TODO: skip volumesnapshotclasses that are from external mode if both internal & external mode is enabled
 		vsc := &volumeSnapshotClassesInCluster.Items[idx]
-		if slices.Contains(supportedCsiDrivers, vsc.Driver) {
+		_, aliasOrRename := destToSrcMap[vsc.Name]
```

Comment:
This change is wrong. a rename only applies to the destination not to the source! You don't select based on renames